Updating assignment (#14)
chamatheapp/chama-frontend-assignment: README.md

<p align="center">
<img src="https://avatars0.githubusercontent.com/u/26920098?s=200&v=4" width="100" height="100" />
</p>

# Frontend Assigment: Github username searcher

## Goal

> The goal of this assignment is to have the candidate work with the same tools that we use every-day on Chama. It is designed to check your coding and problem-solving skills.

Reviewer: ```suggestion
The goal of this assignment is to have the candidate to work with some of the tools that we use every-day on Chama. It is designed to check your coding and problem-solving skills.
```

## Assigment Description

For this assigment, you should consume the public [Github REST API](https://docs.github.com/en/free-pro-team@latest/rest/overview/endpoints-available-for-github-apps#users) in order to present the github data of a given user.

The web app must have two pages:

> - <b>Search page</b>: the user types a github username in a form. Once the form is submitted, the data is fetched and presented in the UI. You can come up showing whatever you want, though it is required to show the user profile metadata (example: avatar picture, name, email, bio, etc.) and his repositories metadata info (example: name, description, URL, etc.)

Reviewer: ```suggestion
- <b>Search page</b>: the user types a Github username in a form. Once the form is submitted, the data is fetched and presented in the UI. You can come up showing whatever you want, though it is required to show the user profile metadata (example: avatar picture, name, email, bio, etc.) and his repositories metadata info (example: name, description, URL, etc.)
```

> - <b>History page</b>: the page must show all previous terms searched by the user, sorted by timestamp. The most recent ones must be on the top. The user must be able to click in the history item and search again using the searched term

Reviewer: Maybe adding something like _"if users close and open the browser, the history must be shown"_ or so

Author: Added!

Share your solution via URL to your git repository or send us a zip file with the source.

## Stack

In your assigment, you are required to use:
- React
- Redux
- Typescript <b>(optional)</b>

## Review

During the assessment, we will look into the following points:

* Whether you use Redux, React, HTML and CSS properly
* Writing testable code
* Code organization (modularity, dependencies between modules, naming, etc)
* SOLID principles
* Error handling

> * Assignment completion

Reviewer: Mobile friendly?

## Hints

* You don't need to spend time creating a dev/build environment. You can use [create-react-app](https://github.com/facebookincubator/create-react-app) (and other alike tools) for that

> * Third-party libraries are allowed, although it is not recommended to use, since our main focus is to assess your skills with HTML, Javascript and CSS

Reviewer: Maybe changing the tone here to something like 
```suggestion
* Third-party libraries are allowed, although you will have extra points if you provide a vanilla solution since our main focus is to assess your skills with HTML, Javascript and CSS
```

## Assigment Description

For this assigment, you should consume the public [Github REST API](https://docs.github.com/en/free-pro-team@latest/rest/overview/endpoints-available-for-github-apps#users) in order to present the github data of a given user.

The web app must have two pages:

- <b>Search page</b>: the user types a Github username in a form. Once the form is submitted, the data is fetched and presented in the UI. You can come up showing whatever you want, though it is required to show the user profile metadata (example: avatar picture, name, email, bio, etc.) and his repositories metadata info (example: name, description, URL, etc.)

- <b>History page</b>: the page must show all previous terms searched by the user, sorted by timestamp. The most recent ones must be on the top. The user must be able to click in the history item and search again using the searched term. If the user close and open the browser, the history must be shown.

Share your solution via URL to your git repository or send us a zip file with the source.

## Stack

In your assigment, you are required to use:
- React
- Redux

> - Typescript <b>(optional)</b>

Reviewer 2: Very small, but does it make sense to say "for bonus points" here instead of "optional"?
It's not a requirement, but it gives us quite some benefit if people know it already or try it. I see going typed as a motivation to think better about naming, separation of concerns and if people apply that well... for sure would be a bonus.

Author: Totally agree. I will update it. thanks, man!
